packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -145,11 +213,25 @@ class BlockListBlock extends Component {
 					accessible={ ! isSelected }
 					accessibilityRole={ 'button' }
 				>
-					<View style={ [ styles.blockHolder, borderStyle, { borderColor } ] }>
+					<View style={ [
+						styles.blockHolder,
+						borderStyle,
+						isSelected && ! isMediaTextParent && ( hasInnerBlock || isInnerBlock || isNestedInnerBlock ) && styles.outlineBorderMargin,
+						isDashed && ! isMediaTextParent && styles.dashedBorderMargin,
+						hasInnerBlock && styles.verticalPaddingNone,
+						{ borderColor },
+					]
+					}>
 						{ showTitle && this.renderBlockTitle() }
 						<View
+							pointerEvents={ isTouchable ? 'auto' : 'box-only' }
 							accessibilityLabel={ accessibilityLabel }
-							style={ [ ! isSelected && styles.blockContainer, isSelected && styles.blockContainerFocused ] }
+							style={ [
+								! isSelected && this.applyUnSelectedStyle(),
+								isSelected && this.applySelectedStyle(),
```

Comment:
I know there are more changes coming here, but generally I'd prefer if we had one method that had all the styling logic and returned a single style object, rather than having these lists of conditionals.

Reply:
Will try to cover that this way and will ping you if anything came up during refactor

Reply:
with latest changes it should be sovled